### PR TITLE
changing traincascade to use the cmdlineparser interface

### PR DESCRIPTION
**Pull request by StevenPuttemans**
_work in progress_

### CODE REVIEW

**Comment by StevenPuttemans** [comment]
@berak / @mshabunin / @alalek as you can see I am switching the traincascade interface, just like the annotation and visualisation tool before, to the new CommandLineParser interface. However I got stuck in the code. Would you guys take a look at the issue and help me get a pointer on how to solve this?

File: apps/traincascade/traincascade.cpp [hunk 1]
```diff
@@ -7,91 +7,64 @@ using namespace cv;
...
-        else if( !strcmp( argv[i], "-numThreads" ) )
-        {
-          numThreads = atoi(argv[++i]);
-        }
-        else if( !strcmp( argv[i], "-acceptanceRatioBreakValue" ) )
-        {
-          acceptanceRatioBreakValue = atof(argv[++i]);
-        }
+
+// STILL NEED TO TRANSLATE THIS
```
Comment (StevenPuttemans) [inline apps/traincascade/traincascade.cpp]: Problem lies here, so replacing the part of the traincascade specific parameters is working, however it has a cascadeParams and stageParams check. How do I cope with this? Do I transfer everything to this file? Or can we use the CommandLineParser over different files? Like the scanAttr function?

**Comment by berak** [comment]
wasps nest. long stick required. prepare to run.

**Comment by StevenPuttemans** [comment]
@berak haha, I get notifiers by mail, deleting it was not the trick ^_^

File: apps/traincascade/traincascade.cpp (hunk 1 above)
Comment (mshabunin) [inline apps/traincascade/traincascade.cpp]: I suggest making several classes with same interface to be used with CommandLineParser:
```.cpp
struct IMyParser {
    virtual std::string getCfgString() const = 0;
    virtual bool initialize(const /* ? */ CommandLineParser &parser) = 0;
};

int main() {
    // ...
    CommandLineParser parser(argc, argv, mainParams.getCfgString() + cascadeParams.getCfgString() /*+ ...*/);
    if (!mainParams.initialize(parser))
        return -1;
    if (!cascadeParams.initialize(parser))
        return -1;
    // ...
}
```

Comment (StevenPuttemans) [inline apps/traincascade/traincascade.cpp]: Hi @mshabunin, I will look into this once I can get the previous change fixed. Could you take a look at https://github.com/opencv/opencv/pull/7004#issuecomment-236166055?

**Pull request closed without merging**